pyresample/utils.py

```diff
@@ -124,53 +146,60 @@ def _parse_yaml_area_file(area_file_name, *regions):
     area_list = regions or area_dict.keys()
-
     res = []
-
     for area_name in area_list:
-        try:
-            params = area_dict[area_name]
-        except KeyError:
-            raise AreaNotFound('Area "{0}" not found in file "{1}"'.format(
-                area_name, area_file_name))
-        description = params['description']
-        projection = params['projection']
-        optimize_projection = params.get('optimize_projection', False)
-        try:
-            xsize = params['shape']['width']
-            ysize = params['shape']['height']
-        except KeyError:
-            xsize, ysize = None, None
-        try:
-            area_extent = (params['area_extent']['lower_left_xy'] +
-                           params['area_extent']['upper_right_xy'])
-        except KeyError:
-            area_extent = None
-        try:
-            projection['units'] = params['area_extent']['units']
-        except KeyError:
-            pass
-        try:
-            rotation = params['rotation']
-        except KeyError:
-            rotation = 0
-        area = DynamicAreaDefinition(area_name, description,
-                                     projection, xsize, ysize,
-                                     area_extent,
-                                     optimize_projection,
-                                     rotation)
-        try:
-            area = area.freeze()
-        except (TypeError, AttributeError):
-            pass
-
-        res.append(area)
+        params = area_dict.get(area_name)
+        if params is None:
+            raise AreaNotFound('Area "{0}" not found in file "{1}"'.format(area_name, area_file_name))
+        params.setdefault('area_id', area_name)
+        # Optional arguments.
+        params['shape'] = _get_list(params, 'shape', ['height', 'width'])
+        params['upper_left_extent'] = _get_list(params, 'upper_left_extent', ['x', 'y'])
+        params['center'] = _get_list(params, 'center', ['x', 'y'])
+        params['area_extent'] = _get_list(params, 'area_extent', ['lower_left_xy', 'upper_right_xy'])
+        params['resolution'] = _get_list(params, 'resolution', ['dx', 'dy'])
+        params['radius'] = _get_list(params, 'radius', ['dx', 'dy'])
+        params['rotation'] = _get_list(params, 'rotation', [])
+        res.append(from_params(**params))
     return res
 
 
+def _get_list(params, var, arg_list):
+    """Reads a list-like param variable."""
+    # Check if variable is in yaml.
+    variable = params.get(var)
+    if not isinstance(variable, dict):
+        return variable
+    list_of_values = []
+    # Add var as a key in case users want to express the entire variable with units. Insert
+    # at 0 to allow Attribute errors when too many arguments are specified.
+    arg_list.insert(0, var)
+    # Iterate through dict.
+    for key in variable.keys():
+        if key not in arg_list and (var == 'shape' or key != 'units'):
```

Comment:
I'm not understanding this, aren't area defs allowed to have a shape ?

Reply:
If I'm understanding this, this chunk of code only happens if the yaml for this parameter is a dictionary. If the key in the yaml sub-dict isn't `units` and it isn't what we asked for then that's an invalid argument. The `shape` check (I think) is because `shape` can't have `units`, it doesn't make sense. @wroberts4 could you add a comment on this?